### DISCUSSION TITLE
[apex] Suppress violations for a class but not for methods in it

### DISCUSSION THREAD

**Discussion by alexmodis** [pmd/pmd#3416] [Q&A] [answered]
```
The class 'XXX' has a total cognitive complexity of 56 (highest 14). (rule: Design-CognitiveComplexity)
```

I have this PMD error that I want to throw out but only for the class part, meaning that if a method in this class raise the same error I want to keep it.

**Comment by oowekyala** [comment, marked as answer]
Does [comment-based suppression](https://pmd.github.io/latest/pmd_userdocs_suppressing_warnings.html#nopmd-comment) help?
```java
class Foo { // NOPMD
}
```
Is your language Java or Apex? In Apex it's only supported since pmd 6.22.0.

**Comment by alexmodis** [comment]
Apex. We are one PMD 6.34.0.
Work like a charm with comment at the end of the line.
Problem come when autoformatter will split the line if too long but that's more related to formatter than PMD itself.
Thanks
